core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -80,6 +80,7 @@ User grantRolesToUser(String metalake, List<String> roles, String user) {
               AuthorizationUtils.ofUser(metalake, user),
               UserEntity.class,
               Entity.EntityType.USER,
+              roles,
               userEntity -> {
                 List<RoleEntity> roleEntities = Lists.newArrayList();
```

Comment:
We can move logic to here.
```  
 roles.forEach(
        roleName -> {
          Namespace namespaceRole = NamespaceUtil.ofRole(ident.namespace().level(0));
          NameIdentifier nameIdentifierRole = NameIdentifier.of(namespaceRole, roleName);
          cache.invalidate(nameIdentifierRole, Entity.EntityType.ROLE);
        });
```

Reply:
Another point, this will execute cache operation multiple times. It's not  an atomic operation.